src/Orleans/Runtime/RequestContext.cs

```diff
@@ -112,19 +113,23 @@ public static void Import(Dictionary<string, object> contextData)
                 if (contextData == null || !contextData.TryGetValue(E2_E_TRACING_ACTIVITY_ID_HEADER, out activityIdObj))
                 {
                     activityIdObj = Guid.Empty;
-                }
 
-#if NETSTANDARD
-                ActivityId.Value = (Guid) activityIdObj;
-#else
-                Trace.CorrelationManager.ActivityId = (Guid)activityIdObj;
+                    object legacyActivityIdObj;
+#if !NETSTANDARD
+                    if (contextData.TryGetValue(E2_E_TRACING_LEGACY_ACTIVITY_ID_HEADER, out legacyActivityIdObj))
+                    {
+                        Trace.CorrelationManager.ActivityId = (Guid)legacyActivityIdObj;
+                    }
 #endif
+                }
+
+                ActivityId.Value = (Guid)activityIdObj;
```

Comment:
Most likely this is a non-issue, but could this cause unnecessary perf impact if the user is never setting the ActivityId in his code (but we set it always to Guid.Empty)? Probably not a concern in terms of memory pressure, but not sure about how expensive is to lift this value when scheduling tasks.
/cc @ReubenBond